cpp/src/interop/decimal_conversion_utilities.cuh

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/column/column_view.hpp>
+#include <cudf/detail/utilities/integer_utils.hpp>
+#include <cudf/detail/utilities/linked_column.hpp>
+#include <cudf/fixed_point/fixed_point.hpp>
+#include <cudf/types.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/device_uvector.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <thrust/for_each.h>
+
+#include <type_traits>
+
+namespace cudf::detail {
+
+namespace {
+
+/**
+ * @copydoc cudf::detail::convert_decimal_data_to_decimal128
+ */
+template <typename DecimalType>
+rmm::device_uvector<__int128_t> gpu_convert_decimal_data_to_decimal128(
```

Comment:
Suggestions welcome if the function name doesn't look the best.

Reply:
I was wondering how would it look with the `gpu` at the end, like `convert_decimal_data_to_decimal128_gpu` ?

Reply:
`gpu` is generally used as a prefix to differentiate right away.

Reply:
I don't think it needs the gpu prefix at all :D
The parameters and the return type are indicative enough IMO.

Reply:
Ah, it's to differentiate from the host code. Duh. 
We usually add `_kernel`, but I'm fine with any naming. This one is not called outside of this file anyway.